source/beamline/CLS/CLSShutters.cpp

```diff
@@ -1,11 +1,16 @@
 #include "CLSShutters.h"
 
 #include "beamline/AMControl.h"
+#include "actions3/AMActionSupport.h"
 #include "actions3/AMListAction3.h"
 
+#include "util/AMErrorMonitor.h"
+
 CLSShutters::CLSShutters(const QString &name, QObject *parent) :
 	CLSTriStateGroup(name, parent)
 {
+	safetyShutter_ = 0;
```

Comment:
In that case, we should change that to not be safety shutter, but rather something like `addReadOnlyComponent` where the readonly components are important for the action to take place but cannot be done programmatically.